lib/msf/core/exploit/smb/client/psexec.rb

```diff
@@ -269,10 +269,7 @@ def execute_powershell_payload
     end
   end
 
-  def native_upload(smb_share)
-    filename = "#{Rex::Text.rand_text_alpha(8)}.exe"
-    serviceencoder = ''
-
+  def native_upload(smb_share, filename, service_encoder)
```

Comment:
I wasn't sure about this change at first, but both `SERVICE_FILENAME` and `SERVICE_STUB_ENCODER` are registered in the modules, so it's fine. Looks like you caught all the uses, too.